### PR TITLE
ARC-1519 - Improve Jira Routing for connect/app creation

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- The combination of the following PRs:
  - https://github.com/atlassian/github-for-jira/pull/1415
  - https://github.com/atlassian/github-for-jira/pull/1417
  - https://github.com/atlassian/github-for-jira/pull/1418
  - https://github.com/atlassian/github-for-jira/pull/1419
  - https://github.com/atlassian/github-for-jira/pull/1420
  - https://github.com/atlassian/github-for-jira/pull/1421
  - https://github.com/atlassian/github-for-jira/pull/1422
  - https://github.com/atlassian/github-for-jira/pull/1423
  - https://github.com/atlassian/github-for-jira/pull/1424
  - https://github.com/atlassian/github-for-jira/pull/1425
  - https://github.com/atlassian/github-for-jira/pull/1449
- Added/Updated routes fixes for 
  - **Connect new server** in _jira-select-server.hbs_
  - **Connect new application** in _jira-select-github-cloud.hbs_
  - Query string corrected in _jira-server-url.hbs_
  - Route added for _Manual app_ in `jira-select-app-creation.hbs`
- Updated the `POCO` `main.json` 

**Why**
- To test all the new routes combined in one branch.

**How has this been tested?**  
- In local environment
- In staging environment - WIP
- Test cases

### CODE REVIEW

**Review by gxueatlassian — COMMENTED** [review]
I am sure all sub PRs are already approved. Let me know if you need me to approve this one, but since I wasn't highly involved in the whole journey, I reckon please wait for approval from @mboudreau or @rachellerathbone.

**Comment by krazziekay** [comment]
> I am sure all sub PRs are already approved. Let me know if you need me to approve this one, but since I wasn't highly involved in the whole journey, I reckon please wait for approval from @mboudreau or @rachellerathbone.

Yup, its all approved already. I've merged it all and created this PR so that it can all be tested and merged in one go.

**Review by mboudreau — COMMENTED** [review]
We need to fully test this in staging...

**Review by rachellerathbone — APPROVED** [review]

**Pull request merged**
